### DISCUSSION TITLE
Print resume error (after filament runout and reload)

### DISCUSSION THREAD

**Discussion by SvenEbens** [Guilouz/Creality-Helper-Script-Wiki#436] [Issues]
Running on a K1 Max with mainsail and the scripts updated a few days ago (have updated the script again after this problem).

I received the following errors (in klipper/mainsail) when I tried to resume a print after the filament ran out and using the display menu to load up another roll.
`Unknown gcode_macro variable 'fan0_speed'`
`{"code":"key165", "msg": "Error evaluating 'gcode_macro RESUME:gcode': jinja2.exceptions.UndefinedError: 'dict object' has no attribute 'fan0_speed'
", "values": ["gcode_macro RESUME:gcode", "jinja2.exceptions.UndefinedError: 'dict object' has no attribute 'fan0_speed'
"]}`
`{"code":"key165", "msg": "Error evaluating 'gcode_macro RESUME:gcode': jinja2.exceptions.UndefinedError: 'dict object' has no attribute 'fan0_speed'`

After restarting I was able to resume, however unfortunately the nozzle started higher than it was supposed to

I attached the klippy log as well.
[klippy.log](https://github.com/Guilouz/Creality-K1-Series/files/14605711/klippy.log)

Any idea what the problem is? I do see the fan0speed stuff in M600 support config, but not sure what the problem is.

**Comment by Guilouz** [comment]
Do you have updated script before, i have updated M600 macros?

**Reply by SvenEbens** [reply]
Ah alright so probably fixed already in the latest?

I updated the scripts and did a factory reset of cfg's 4 days ago (when 5.0 came out). Then ran into this issue today (roll ran out) and now (so after the issue occurred) I updated the script to 5.0.1

Does the M600 config update with that as well or should I do another factory reset (or uninstall/reinstall of the M600 support to get the newest version)?  (BTW super thanks for your great support on these printers, makes them so much more useful!)

**Comment by Guilouz** [comment]
Update to latest script update and your m600 will be updated

**Comment by SvenEbens** [comment]
Great then I guess it should indeed be fixed now. Wasn't sure it was something that was addressed in the 5.0.0>5.0.1 so thought I would post it anyways.

**Comment by ElForastero** [comment]
Just got this error on a v5 while trying to use multicolor printing.

<img width="750" alt="image" src="https://github.com/Guilouz/Creality-K1-Series/assets/5102818/edcd4d35-c80b-449c-84ef-c51b266fe754">

**Reply by Guilouz** [reply]
Do you have lastest script update?

**Reply by ElForastero** [reply]
Oh, I see you have already updated the macro a couple of days ago, so I just need to update.
Thank you 🙏

**Reply by Guilouz** [reply]
Is it ok now ?

**Reply by ElForastero** [reply]
Yes, everything is fine now. Works as expected 👍
